vmware_aria_operations_integration_sdk/adapter_configurations/adapter_templates/java/sample_adapter/src/main/java/Adapter.java.template

```diff
@@ -1,100 +1,324 @@
         }
+
+        return result;
     }
 
-    private static JsonNode getContent(String pipe) {
-        JsonNode map = null;
+    public CollectResult collect(AdapterInstance adapterInstance) {
+        Timer timer = new Timer("Collection");
+        Logger logger = AdapterLogger.getLogger();
+        CollectResult result = new CollectResult();
 
-        try {
-            FileInputStream inPipe = new FileInputStream(pipe);
-            ObjectMapper mapper = new ObjectMapper();
+        try{
+            // Sample collection code follows. Replace this with your own collection
+            // code. A typical collection will generally consist of:
+            // 1. Read identifier values from adapter_instance that are required to
+            //    connect to the target(s)
+            // 2. Connect to the target(s), and retrieve data
+            // 3. Add the data into a CollectResult's objects, properties, metrics, etc
+            // 4. Disconnect cleanly from the target (ensure this happens even if an
+            //    error occurs)
+            // 5. Return the CollectResult.
+
+
+            // CPU
+            OperatingSystemMXBean jVMManagementInterface = ManagementFactory.getPlatformMXBean(OperatingSystemMXBean.class);
+            int availableProcessors = jVMManagementInterface.getAvailableProcessors();
+            logger.info("Number of CPUs: " + availableProcessors);
+            Object cpu = result.getOrCreateObject(ADAPTER_KIND, "cpu", "CPU");
+            cpu.withProperty("cpu_count", availableProcessors);
+
+            SystemInfo systemInformation = new SystemInfo();
+
+            CentralProcessor processor = systemInformation.getHardware().getProcessor();
+
+            // Get an array of CPU tick times
+            long[] startTicks = processor.getSystemCpuLoadTicks();
+            Thread.sleep(1000);
+            long[] endTicks = processor.getSystemCpuLoadTicks();
+
+            // Calculate the CPU times
+            long userTime = endTicks[TickType.USER.getIndex()] - startTicks[TickType.USER.getIndex()];
+            long systemTime = endTicks[TickType.SYSTEM.getIndex()] - startTicks[TickType.SYSTEM.getIndex()];
+            long idleTime = endTicks[TickType.IDLE.getIndex()] - startTicks[TickType.IDLE.getIndex()];
 
-            map = mapper.readTree(inPipe);
+            cpu.withMetric("user_time", userTime);
+            cpu.withMetric("system_time", systemTime);
+            cpu.withMetric("idle_time", idleTime);
 
-        } catch (IOException e){
-                System.err.println("Error reading/writing to the pipes: " + e.getMessage());
-                e.printStackTrace();
+            // Disks
+            ArrayList<Object> disks = new ArrayList<>();
+            for (Path root : FileSystems.getDefault().getRootDirectories()) {
+                try {
+                    FileStore store = Files.getFileStore(root);
+                    Object disk = result.getOrCreateObject(ADAPTER_KIND, "disk", "Disk");
```

Comment:
The third argument is `name`. Since we don't have any identifiers, if there are more than one root directories we will be adding all the metrics to the same object. To fix, probably the easiest thing to do would be use `root.toString` for the name and drop the `partition` property (as it is now redundant.)